src/Session.js

```diff
@@ -79,8 +89,9 @@ const Session = class Session {
 
     query(querySpec) {
         const { table } = querySpec;
-        this.markAccessed(table);
-        return this.db.query(querySpec, this.state);
+        const result = this.db.query(querySpec, this.state);
+        this.markAccessed(table, result.rows.map(r => r.id));
```

Comment:
I don't know how to get the model's `idAttribute`, but this should reference whatever that is instead of just ".id". 

I also don't know if it's safe to peer into the query result in this particular way?

Reply:
r.id => getId() ?

Reply:
you can also try `this[table].idAttribute`